hypothesis-python/src/hypothesis/internal/observability.py

```diff
@@ -231,6 +356,15 @@ def _system_metadata() -> dict[str, Any]:
 OBSERVABILITY_COLLECT_COVERAGE = (
     "HYPOTHESIS_EXPERIMENTAL_OBSERVABILITY_NOCOVER" not in os.environ
 )
+#: If ``True``, include the ``metadata.choice_nodes`` key in test case
+#: observations.
+#:
+#: ``False`` by default. ``metadata.choice_nodes`` can be substantial amount of
+#: data, and so must be opted-in to, even when observability is enabled.
+OBSERVABILITY_CHOICE_NODES = (
+    "HYPOTHESIS_EXPERIMENTAL_OBSERVABILITY_CHOICE_NODES" in os.environ
+)
```

Comment:
also needs the experimental / unstable note, precisely because I'm keen to find a better interface for this.